eldritch/events.py

```diff
@@ -612,6 +612,87 @@ def finish_str(self):
     return f"{self.character.name} keeps " + ", ".join(self.kept)
 
 
+class Sell(Event):
+
+  def __init__(self, character, target_decks, sell_count=1, discount_type="fixed", discount=0,
+               prompt="Sell items?"):
+    assert discount_type in {"fixed", "rate"}
+    self.character = character
+    self.prompt = prompt
+    self.sell_count = sell_count
+    self.sellable = None
+    self.discount_type = discount_type
+    self.discount = discount
+    self.choice = None
+    self.prices = None
+    self.sold = []
+    self.target_decks = target_decks
+    self.resolved = False
+
+  def resolve(self, state):
+    if self.resolved:
+      # This should never happen??
+      return True
+
+    if self.choice is not None:
+      if not self.choice.is_resolved():  # This should never happen
+        return False
+      if self.choice.choices[self.choice.choice_index] == 'Nothing':
+        self.resolved = True
+        return True
+
+      card = self.sellable.pop(self.choice.choice_index)
+      price = self.prices.pop(self.choice.choice_index)
+
+      self.character.possessions.remove(card)
+      getattr(state, card.deck).append(card)
+
+      self.character.dollars += price
+
+      self.sell_count -= 1
+
+    if self.sellable is None:
+      self.sellable = []
+      for item in self.character.possessions:
+        if item.deck in self.target_decks:
+          self.sellable.append(item)
+
+    if self.sell_count > 0:
+      choices = []
+      self.prices = []
+      for card in self.sellable:
+        price = self.discounted_price(card)
+        self.prices.append(price)
+        choices.append(f"{card.name} for ${price}")
```

Comment:
In order for CardChoice to display properly in the UI, the choice names have to be the actual card names. But that shouldn't be a problem if you switch to ItemChoice (see above).